tests/e2e/replication/show.py

```diff
@@ -27,17 +32,49 @@ def test_show_replication_role(port, role, connection):
 
 
 def test_show_replicas(connection):
+    # Goal of this test is to check the SHOW REPLICAS command.
+    # 1/ We check that all replicas have the correct state: they should all be ready.
+    # 2/ We drop one replica. It should not appear anymore in the SHOW REPLICAS command.
+    # 3/ We kill another replica. It should become invalid in the SHOW REPLICAS command.
     cursor = connection(7687, "main").cursor()
+    EXPECTED_COLUMN_NAMES = {"name", "socket_address", "sync_mode", "timeout", "state"}
+
+    # 1/
     actual_data = set(execute_and_fetch_all(cursor, "SHOW REPLICAS;"))
 
-    expected_column_names = {"name", "socket_address", "sync_mode", "timeout"}
+    expected_column_names = {"name", "socket_address", "sync_mode", "timeout", "state"}
     actual_column_names = {x.name for x in cursor.description}
-    assert expected_column_names == actual_column_names
+    assert EXPECTED_COLUMN_NAMES == actual_column_names
+    expected_data = {
+        ("replica_1", "127.0.0.1:10001", "sync", 0, "ready"),
+        ("replica_2", "127.0.0.1:10002", "sync", 1.0, "ready"),
+        ("replica_3", "127.0.0.1:10003", "async", None, "ready"),
+    }
+    assert expected_data == actual_data
+
+    # 2/
+    execute_and_fetch_all(cursor, "DROP REPLICA replica_2")
+    actual_data = set(execute_and_fetch_all(cursor, "SHOW REPLICAS;"))
+    expected_data = {
+        ("replica_1", "127.0.0.1:10001", "sync", 0, "ready"),
+        ("replica_3", "127.0.0.1:10003", "async", None, "ready"),
+    }
+    assert expected_data == actual_data
+
+    # 3/
+    for line in os.popen("ps ax | grep -E '7688|7690' | grep -E 'replica1.log|replica3.log' | grep -v grep"):
```

Comment:
I want opinion on that one. Here I want to kill the replicas and check that the main realizes it and mark their state as invalid.
I came up with this solution but am not completely convinced by it. Works on Linux, can't ensure for other OS. 
Here are the other options I have though about:

- When setting up the test from the workload file/runner.py, keeping the PID of MemgraphInstances corresponding to the replicas in a global variable to kill them afterwards. However given the setup and fact that a python script call a bash script which call python etc... This is not possible/easy.

- When setting up the test from the workload file/runner.py, passing instances of MemgraphInstances corresponding to the replicas to the test file. However given the setup and fact that a python script call a bash script which call python etc... This is not possible/easy.

- Writing a new test file (show2.py, but better naming) where I manually start the replicas myself then kill them. I think this is the "best" but requires more devel time.

Reply:
I think there is a general way to do this with `[psutil](https://pypi.org/project/psutil/)`. It should be something like:

`
for conn in filter(lambda conn: conn.laddr and conn.laddr.port == 7688, psutil.net_connections()):
  os.kill(con.pid, signal.SIGKILL)
`
I haven't tested the above, if it doesn't work, it should be simple to modify based on the docs :)

Reply:
I will give it a try

Reply:
@kostasrim as we discussed I have tried the approach where I start the instances myself. It works as is and then we do not risk to kill something we are not supposed to.